app/helpers/task_view_helper.rb

```diff
@@ -1,18 +1,43 @@
 module TaskViewHelper
   def task_view_header(choice)
+    yield (case choice&.task_view_group
+           when 1 then 'Received over 30 days ago - make a decision now'
+           when 2 then 'Received – make a decision'
+           when 3 then 'Confirm deferred offers'
+           when 5 then 'Give feedback: you did not make a decision in time'
+           when 6 then 'Interviewing'
+           when 7 then 'Offers pending conditions (previous cycle)'
+           when 8 then 'Waiting for candidate to respond to offer'
+           when 9 then 'Offers pending conditions (current cycle)'
+           when 10 then 'Successful candidates'
+           when 11 then 'Deferred offers'
+           else
+             'No action needed'
+           end)
+  end
+
+  def display_header?(application_choices, choice)
+    return true if (index = application_choices.index(choice)).zero?
+
+    choice.task_view_group != application_choices[index - 1].task_view_group
+  end
+
+  def relative_date_text_color(choice)
+    return unless choice.respond_to?(:task_view_group)
+
     case choice&.task_view_group
```

Comment:
Do we need both `respond_to?(:task_view_group)` and `&.task_view_group`?
Are we expecting more color coding? If that's not the case I'm not sure this justifies a `case`.

Reply:
The safe navigation is indeed not needed. I went with case for consistency across the methods.